shared_model/validators/protobuf/proto_transaction_validator.hpp

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_PROTO_TRANSACTION_VALIDATOR_HPP
+#define IROHA_PROTO_TRANSACTION_VALIDATOR_HPP
+
+#include "validators/transaction_validator.hpp"
+
+namespace shared_model {
+  namespace validation {
+
+    template <typename FieldValidator, typename CommandValidator>
+    class ProtoTransactionValidator
+        : public TransactionValidator<FieldValidator, CommandValidator> {
+     private:
+      void validateProtoTx(const iroha::protocol::Transaction &transaction,
+                           ReasonsGroupType &reason) const {
+        for (const auto &command :
+             transaction.payload().reduced_payload().commands()) {
+          if (command.command_case()
```

Comment:
Consider using switch-case

Reply:
Already did. Then I should process every command, otherwise warning would be thrown

Reply:
ehm, what about `default`?